src/c_ext/hamt/hamt.hpp

```diff
@@ -0,0 +1,165 @@
+/**
+ * inspired by http://blog.mattbierner.com/hash-array-mapped-tries-in-javascript/
+ */
+
+#include <iostream>
+#include <string>
+#include <exception>
+#include <vector>
+#include <queue>
+
+#include "node.hpp"
+
+template <typename T>
+class HAMT {
+private:
+    int size;
+    hamt::node<T> root;
+
+public:
+    HAMT(const int size, const hamt::node<T> root)
```

Comment:
- nit: `size` doesn't need to be const, since it's passed by value
- should `root` be passed by `const ref`? in this case, `&` is missing.